docs/pages/docs/collaboration/comments.mdx

```diff
@@ -0,0 +1,138 @@
+
+_Note: Because writes are executed via a REST API, the `RESTYjsThreadStore` is not suitable for local-first applications that should be able to add and edit comments offline._
+
+### `TiptapThreadStore`
+
+The `TiptapThreadStore` integrates with Tiptap's collaboration provider for comment management. This implementation is designed specifically for use with Tiptap's collaborative editing features.
+
+```tsx
+import { TiptapThreadStore, DefaultThreadStoreAuth } from "@blocknote/core";
+import { TiptapCollabProvider } from "@hocuspocus/provider";
+
+// Create a TiptapCollabProvider (you probably have this already)
+const provider = new TiptapCollabProvider({
+  name: "test",
+  baseUrl: "https://collab.yourdomain.com",
+  appId: "test",
+  document: doc,
+});
+
+// Create a TiptapThreadStore
+const threadStore = new TiptapThreadStore(
+  userId, // The active user's ID
+  provider, // Tiptap collaboration provider
+  new DefaultThreadStoreAuth(userId, "editor"), // Authorization rules (see below)
+);
+```
+
+### ThreadStoreAuth
+
+The `ThreadStoreAuth` class defines the authorization rules for interacting with comments. Every ThreadStore implementation requires a `ThreadStoreAuth` instance. BlockNote uses the `ThreadStoreAuth` instance to deterine which interactions are allowed for the current user (for example, whether they can create a new comment, edit or delete a comment, etc.).
+
+The `DefaultThreadStoreAuth` class provides a basic implementation of the `ThreadStoreAuth` class. It takes a user ID and a role ("comment" or "editor") and implements the rules. See the [source code](https://github.com/TypeCellOS/BlockNote/blob/main/packages/core/src/extensions/Comments/threadstore/DefaultThreadStoreAuth.ts) for more details.
+
+_Note: The `ThreadStoreAuth` only used to show / hide options in the UI. To secure comment related data, you still need to implement your own server-side validation (e.g. using `RESTYjsThreadStore` and a secure REST API)._
+
+## `resolveUsers` function
+
+When a user interacts with a comment, the data is stored in the ThreadStore, along with the active user ID (as specified when initiating the ThreadStore).
+
+To display comments, BlockNote needs to retrieve user information (such as the username and avatar) based on the user ID. To do this, you need to provide a `resolveUsers` function in the editor options.
+
+This function is called with an array of user IDs, and should return an array of `User` objects in the same order.
+
+```tsx
+type User = {
+  id: string;
+  username: string;
+  avatarUrl: string;
+};
+
+async function myResolveUsers(userIds: string[]): Promise<User[]> {
+  // fetch user information from your database / backend
+  // and return an array of User objects
+
+  return await callYourBackend(userIds); //
+
+  // Return a list of users
+  return users;
+}
+```
```

Comment:
Maybe this should be a `Map<User['id'], User>` to make it order-independent and automatically de-duped

Reply:
Thanks. Might be a good idea, agree it's a better API with a Map, but also a bit more complex to explain maybe.

I'll decide when finalizing unless you feel very strong about it :)